source/linux/Foundational_Components/U-Boot/UG-Memory.rst

```diff
@@ -1,973 +1,1120 @@
 SD, eMMC and USB
 ----------------
 
-The commands for using SD cards, eMMC flash and USB mass storage devices
-(hard drives, flash drives, card readers, etc) are all very similar. The
-biggest difference is that on some hardware we may not be able to run
-U-Boot out of ROM from the storage device as it is unsupported. Once
-U-Boot is running however, any of these may be used for the kernel and
-the root filesystem.
+The following guide shows how to flash and boot from storage media like the
+embedded multimedia card (eMMC), secure digital (SD) card, and USB storage
+devices. While this is a step-by-step guide, it is in no way extensive and
+does not cover all the per-platform corner-cases. For any  issues/questions
+on this guide, please reach out to: `E2E <https://e2e.ti.com//>`__.
+
+MMC
+===
+
+.. _uboot-listing-mmc-devices:
+
+Listing MMC devices
+^^^^^^^^^^^^^^^^^^^
```

Comment:
Any particular reason to jump from == (sections) used for MMC to ^^^ (subsubsections) directly?
The ideal flow should be `#   ==>  *   ==>   =   ==>   -   ==>   ^   ==> " `

https://www.sphinx-doc.org/en/master/usage/restructuredtext/basics.html#sections